improver_tests/cli/test_init.py

```diff
@@ -165,113 +166,94 @@ def test_with_output(self, m):
-
-    @patch("improver.cli.maybe_coerce_with", new=replace_load_with_extract)
+        result = func(self.wind_cubes)
+        self.assertIn(self.wind_speed_cube, result)
+        self.assertIn(self.wind_dir_cube, result)
+        self.assertEqual(2, len(result))
+        mocked_maybe_coerce.assert_called_once()
+
     def test_list_two_valid(self):
-        """Tests that providing two valid constraints raises a ValueError."""
-        obj = create_constrained_inputcubelist_converter(
-            ["wind_speed", "wind_from_direction"]
+        """Tests that one cube is loaded from each list."""
+        func = create_constrained_inputcubelist_converter(
+            lambda cube: cube.name() in ["speed", "wind_speed"],
+            lambda cube: cube.name() in ["dir", "wind_from_direction"],
         )
-        msg = "Incorrect number of valid inputs available"
-        with self.assertRaisesRegex(ValueError, msg):
-            obj(self.wind_cubes)
+        result = func(self.wind_cubes)
+        self.assertIn(self.wind_speed_cube, result)
+        self.assertIn(self.wind_dir_cube, result)
+        self.assertEqual(2, len(result))
+
+    def test_list_two_diff_shapes(self):
+        """Tests that one cube is loaded from each list
+        when the two lists are different sizes.
+        """
+        func = create_constrained_inputcubelist_converter(
+            lambda cube: cube.name() in ["wind_speed"],
+            lambda cube: cube.name() in ["dir", "wind_from_direction"],
+        )
+        result = func(self.wind_cubes)
+        self.assertIn(self.wind_speed_cube, result)
+        self.assertIn(self.wind_dir_cube, result)
+        self.assertEqual(2, len(result))
 
-    @patch("improver.cli.maybe_coerce_with", new=replace_load_with_extract)
     def test_list_no_match(self):
-        """Tests that providing no valid constraints raises a ValueError."""
-        obj = create_constrained_inputcubelist_converter(["nonsense"])
-        msg = "Incorrect number of valid inputs available"
-        with self.assertRaisesRegex(ValueError, msg):
-            obj(self.wind_cubes)
-
-    @patch("improver.cli.maybe_coerce_with", new=replace_load_with_extract)
-    def test_list_one_optional_constraint(self):
-        """Tests that a list returns a function which itself returns 2 cubes"""
-        obj = create_constrained_inputcubelist_converter(
-            ["wind_speed", "nonsense"], "wind_from_direction"
-        )
-        result = obj(self.wind_cubes)
-        self.assertEqual(result, self.wind_cubes)
-
-    @patch("improver.cli.maybe_coerce_with", new=replace_load_with_extract)
-    def test_list_mismatching_lengths(self):
-        """Tests that a list returns a function which itself returns 2 cubes"""
-        obj = create_constrained_inputcubelist_converter(
-            ["wind_speed", "nonsense"], ["wind_from_direction"]
+        """Tests that providing no valid constraints raises a ConstraintMismatchError."""
```

Comment:
It might be useful to add a test for if multiple valid inputs are provided e.g. 
```
        func = create_constrained_inputcubelist_converter(
            lambda cube: cube.name() in ["wind_speed", "wind_from_direction"],
        )
```
This seems to give an error similar to the `test_list_no_match` test.